src/main/java/org/apache/mesos/api/JettyApiServer.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.mesos.api;
+
+import org.apache.mesos.scheduler.plan.api.PlanResource;
+import org.apache.mesos.scheduler.recovery.api.RecoveryResource;
+import org.apache.mesos.state.api.StateResource;
+import org.eclipse.jetty.server.Server;
+import org.glassfish.jersey.jetty.JettyHttpContainerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * A JettyApiServer takes a list of POJO JAX-RS Resources and serves them on the indicated port of 0.0.0.0.
+ */
+public class JettyApiServer {
```

Comment:
@mohitsoni:  Ultimately, it's up to the service author.  If this simple Jetty server is sufficient for their needs, they can use it.  If it's not sufficient for their needs, they can use something else.  Providing a general purpose HTTP server is far outside the scope of this PR, or any PR really, as there are many general purpose HTTP servers available, including Jetty.

The author's choices would boil down to whether they want to use Default implementations of interfaces (which are not yet fully defined).  They could not use the DefaultService implementation.  They could not use the DefaultScheduler implementation.  If either of these layers are customized then the author would be free to use their own HTTP server.

The Default implementations need something, this is at that something.  The SDK is not making an opinionated statement on how HTTP resources should be served.  It is providing something, rather than nothing.